tool.go

```diff
@@ -169,10 +170,9 @@ func main() {
 				}
 				return nil
 			}()
-			exitCode := handleError(err, options, nil)
 
 			if s.Watcher == nil {
-				os.Exit(exitCode)
+				return err
```

Comment:
This is may be correct, but it looks odd.  Checking for `s.Watcher == nil` instead of `err != nil`?

Reply:
I think the idea is that if the watch mode is activated, we don't want to exit upon compiler error, but simply wait for another iteration to happen. However, your comment highlighted another flaw in my code, which I'll fix shortly.